src/main/java/org/jenkinsci/plugins/github_branch_source/GitHubSCMSource.java

```diff
@@ -825,7 +825,7 @@ public SCM build(SCMHead head, SCMRevision revision) {
                 setBrowser(scm, repoUrl);
             }
             return scm;
-        } else if (head instanceof PullRequestSCMHead) {
+        } else if (head instanceof PullRequestSCMHead && ((PullRequestSCMHead) head).getSourceRepo() != null) {
```

Comment:
i believe it is only theoretically positive end up here with the null... but if it does then we will fall through... 

But the git impl doesn't know how to handle pull request heads, so should return null itself... unless you happen to have called your branch PR-123 in which case the PR branch would be masked by the non PR branch with a name using the PR branch name pattern so you couldn't get here anyway.

If I think I follow what I just wrote... probably fine!